pallets/loans/src/rate_model.rs

```diff
@@ -12,16 +12,26 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use crate::*;
 use primitives::{Rate, Ratio, Timestamp, SECONDS_PER_YEAR};
-use sp_runtime::traits::{CheckedAdd, CheckedDiv, CheckedSub, Saturating};
+use sp_runtime::traits::{CheckedAdd, CheckedDiv, CheckedMul, CheckedSub, Saturating};
 
-use crate::*;
+const STAKING_APY: Ratio = Ratio::from_percent(14);
+
+pub trait InterestRateModelLike {
+    /// Check the model for sanity
+    fn check_model(&self) -> bool;
+
+    /// Calculates the borrow interest rate of the model
+    fn get_borrow_rate(&self, utilization: Ratio) -> Option<Rate>;
+}
 
 /// Parallel interest rate model
 #[derive(Encode, Decode, Eq, PartialEq, Copy, Clone, RuntimeDebug)]
 pub enum InterestRateModel {
-    Jump(JumpModel),
     Curve(CurveModel),
+    Jump(JumpModel),
+    Polynomial(PolynomialModel),
```

Comment:
PolynomialModel is still under discussion and should not add to this PR. Please remove this related code and let this PR be merged asap.